jinahub/indexers/searcher/HnswlibSearcher/hnswlib_searcher.py

```diff
@@ -55,9 +57,10 @@ def __init__(
         self.ef_query = ef_query
         self.max_connection = max_connection
         self.dump_path = dump_path
+        self.is_distance = is_distance
 
         self.logger = JinaLogger(self.__class__.__name__)
-        self._index = hnswlib.Index(space=self.metric, dim=self.dim)
+        self._index = hnswlib.Index(space=self.metric_type, dim=self.dim)
```

Comment:
Here you need to catch and change `'l2'` -> `'euclidean'`

Reply:
The mapping logic happens in `metric_type()` property function.